src/hotspot/share/opto/superwordVTransformBuilder.cpp

```diff
@@ -228,7 +228,7 @@ VTransformNode* SuperWordVTransformBuilder::get_or_make_vtnode_vector_input_at_i
       return shift_count;
     } else {
       // Replicate the scalar same_input to every vector element.
-      BasicType element_type = _vloop_analyzer.types().velt_basic_type(p0);
+      BasicType element_type = p0->is_Convert() ? p0->in(1)->bottom_type()->basic_type() : _vloop_analyzer.types().velt_basic_type(p0);
```

Comment:
What vectors are generated (or not) with this change? The array in the test ins `int[]` but the element_type will be Long now. Will it bailout vectorization?

Reply:
Thank you for explanation. Yes, to have comment would be nice.

Reply:
I added a comment. Does it look ok to you?